src/app/screens/Enable/LiquidEnable.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+import LiquidEnableComponent from "~/app/components/Enable/LiquidEnable";
+import Onboard from "~/app/components/onboard";
+import { useAccount } from "~/app/context/AccountContext";
+import api from "~/common/lib/api";
+import type { OriginData } from "~/types";
+
+type Props = {
+  origin: OriginData;
+};
+
+export default function LiquidEnable(props: Props) {
+  const { account } = useAccount();
+  const [hasMnemonic, setHasMnemonic] = useState(false);
+
+  useEffect(() => {
+    async function fetchAccountAndSetComponent() {
+      try {
+        const fetchedAccount = await api.getAccount();
+
+        if (fetchedAccount.hasMnemonic) {
+          setHasMnemonic(true);
+        } else {
+          setHasMnemonic(false);
+        }
+      } catch (e) {
+        console.error(e);
+      }
+    }
+
+    fetchAccountAndSetComponent();
+  }, [props.origin, account]);
+
+  return (
+    <div>
+      {hasMnemonic ? (
+        <LiquidEnableComponent origin={props.origin} />
```

Comment:
Isn't this back to front? now we have duplicated enable components, but the Onboard component (which should be different) is the same across all enable screens

Reply:
Decided to leave this for now